filter/bufferfilter.go

```diff
@@ -76,22 +81,23 @@ func NewBufferFilter(e ast.BooleanExpr) (*BufferFilter, error) {
 			return nil, nil
 		}
 		if e.Value.Type == "string" {
-			// TODO Match the behavior of searchRecordString.
-			return nil, nil
-		}
-		for _, r := range e.Text {
-			// TODO Make stringFinder insensitive to case. Until
-			// then, bail if e.Text contains a letter.
-			if unicode.IsLetter(r) {
+			pattern, err := zng.TypeBstring.Parse([]byte(e.Value.Value))
+			if err != nil {
+				return nil, err
+			}
+			left := newBufferFilterForStringCase(string(pattern))
+			if left == nil {
 				return nil, nil
 			}
+			right := &BufferFilter{
+				op:  opFieldNameFinder,
+				fnf: newFieldNameFinder(string(pattern)),
+			}
+			return &BufferFilter{op: opOr, left: left, right: right}, nil
 		}
-		left := &BufferFilter{
-			op:           opStringFinder,
-			stringFinder: makeStringFinder(e.Text),
-		}
+		left := newBufferFilterForStringCase(e.Text)
```

Comment:
For my edification, what's an example query that takes the `if e.Value.Type == "string" {` branch & an example that doesn't, so gets processed here on line 98 and onwards?. I looked at zql.peg, but the `searchPred` case is complex.

Reply:
`if e.Value.Type == "string" { ` corresponds to a search term that the parser identifies as a string.
```
$ cd zql && go run ./main whatever | sed 1,/Js/d
{
    "op": "FilterProc",
    "filter": {
        "op": "Search",
        "text": "whatever",
        "value": {
            "op": "Literal",
            "type": "string",
            "value": "whatever"
        }
    }
}
```

Line 98 and on handles search terms of any other type.
```
$ cd zql && go run ./main 123 | sed 1,/Js/d
{
    "op": "FilterProc",
    "filter": {
        "op": "Search",
        "text": "123",
        "value": {
            "op": "Literal",
            "type": "int64",
            "value": "123"
        }
    }
}
```